api/namex/resources/requests.py

```diff
@@ -383,6 +378,65 @@ def post(self, *args, **kwargs):
         current_app.logger.info('Someone is trying to post a new request')
         return jsonify({'message': 'Not Implemented'}), 501
 
+# For sbc-auth - My Business Registry page.
+@cors_preflight("POST")
+@api.route('/search', methods=['POST'])
+class RequestSearch(Resource):
+
+    @staticmethod
+    @cors.crossdomain(origin='*')
+    @jwt.has_one_of_roles([User.SYSTEM])
+    def post():
+        search = request.get_json()
+        name = search.get('name', None)
+        identifiers = search.get('identifiers', [])
+        search_identifier = search.get('searchIdentifier', None)
+        status = search.get('status', None)
+        page = search.get('page', 1)
+        limit = search.get('limit', 20)
+        q = RequestDAO.query.filter(RequestDAO.nrNum.in_(identifiers)) \
+            .options(
+                lazyload('*'),
+                eagerload(RequestDAO.names).load_only(Name.state, Name.name),
```

Comment:
I tried to load as few fields as possible. 

Example query after serialization:

```
 SELECT requests.nr_num AS requests_nr_num, requests.request_type_cd AS requests_request_type_cd, requests.nature_business_info AS requests_nature_business_info, requests.state_cd AS requests_state_cd, requests.entity_type_cd AS requests_entity_type_cd, requests.id AS requests_id, names_1.id AS names_1_id, names_1.name AS names_1_name, names_1.state AS names_1_state, applicants_1.party_id AS applicants_1_party_id, applicants_1.phone_number AS applicants_1_phone_number, applicants_1.email_address AS applicants_1_email_address 
FROM requests LEFT OUTER JOIN names AS names_1 ON requests.id = names_1.nr_id LEFT OUTER JOIN applicants AS applicants_1 ON requests.id = applicants_1.nr_id 
WHERE requests.nr_num IN (%(nr_num_1_1)s, %(nr_num_1_2)s, %(nr_num_1_3)s, %(nr_num_1_4)s, %(nr_num_1_5)s) AND (requests.name_search ILIKE %(name_search_1)s OR requests.name_search ILIKE %(name_search_2)s OR requests.name_search ILIKE %(name_search_3)s) AND requests.id IN (SELECT anon_1.id 
FROM (SELECT requests.id AS id 
FROM requests 
WHERE requests.nr_num IN (%(nr_num_1_1)s, %(nr_num_1_2)s, %(nr_num_1_3)s, %(nr_num_1_4)s, %(nr_num_1_5)s) AND (requests.name_search ILIKE %(name_search_1)s OR requests.name_search ILIKE %(name_search_2)s OR requests.name_search ILIKE %(name_search_3)s) GROUP BY requests.id 
 LIMIT %(param_1)s OFFSET %(param_2)s) AS anon_1)
```

There is an index on nr_id already that exists.